products/extensions/b2b-components/index.md

```diff
@@ -20,3 +20,134 @@ In the world of digital B2B commerce, where businesses engage with other compani
+
+### Prerequisite
+
+To improve organization and maintain a clear structure, it is advisable to relocate all B2B Components into the `B2B` folder within the Commercial plugin. By doing so, you can centralize the B2B-related functionality, making it easier to locate, manage, and maintain the codebase. This folder structure promotes better separation of concerns and enhances the overall modularity of the application.
+
+```
+├── src
+│   ├── B2B
+│   │   ├── QuickOrder
+│   │   ├── AnotherB2BComponent
+│   │   │   CommercialB2BBundle.php
+...
+
+```
+
+To ensure consistency and clarity, it is recommended to make your B2B Component extend `CommercialB2BBundle` instead of `CommercialBundle` and add the `type` as **B2B** attribute inside the `describeFeatures()` method of each B2B Component. This attribute will help identify and categorize the features specifically related to B2B functionality.
+
+By including `type => 'B2B'` in the `describeFeatures()` method, you can distinguish B2B features from other types of features within your application. This will facilitate easier maintenance, organization, and identification of B2B-related functionalities, ensuring a streamlined development process.
+
+For example, consider the following code snippet:
+
+```php
+namespace Shopware\Commercial\B2B\YourB2BComponent;
+
+class YourB2BComponent extends CommercialB2BBundle
+{
+    public function describeFeatures(): array
+    {
+        return [
+            [
+                ...,
+                'type' => self::TYPE_B2B,
+            ],
+        ];
+    }
+}
+```
+
+## Using feature toggle in Route/API/Controller
+
+### Using new service
+
+To determine if a customer is allowed to access a specific B2B feature, we will utilize the `isAllowed()` method from the `Shopware\Commercial\B2B\QuickOrder\Domain\CustomerSpecificFeature\CustomerSpecificFeatureService` service. This method accepts two parameters: the customer ID and the technical code of the B2B component.
+
+We will place this check before every route, controller or API as follows:
+
+```php
+use Shopware\Commercial\B2B\QuickOrder\Domain\CustomerSpecificFeature\CustomerSpecificFeatureService;
+ 
+class ApiController
+{
+    public function __construct(private readonly CustomerSpecificFeatureService $customerSpecificFeatureService)
+    {
+    }
+
+    #[Route(
+        path: '/your/path',
+        name: 'path.name',
+        defaults: ['_noStore' => false, '_loginRequired' => true],
+        methods: ['GET'],
+    )]
+    public function view(Request $request, SalesChannelContext $salesChannelContext): Response
+    {
+        if (!$this->customerB2BFeatureService->isAllowed($salesChannelContext->getCustomerId(), 'QUICK_ORDER')) {
+            throw CustomerSpecificFeatureException::notAllowed('QUICK_ORDER');
+        }
+
+        ...
+    }
+```
+
+## Using feature toggle in Twig - Storefront
+
+### New Twig function
```

Comment:
Is this heading necessary to be mentioned ?

Reply:
We can remove it, due the first sentences we said "You can use a new Twig extension called..."